rollup/ui5-resolve.js

```diff
@@ -0,0 +1,118 @@
+const path = require("path");
+const fs = require("fs");
+const stream = require("stream");
+const { promisify } = require("util");
+const pipe = promisify(stream.pipeline);
+
+const convertAMDtoES6 = require("@buxlabs/amd-to-es6");
+const amdextract = require("amdextract");
+
+const log = require("@ui5/logger").getLogger("rollup-plugin-ui5-resolve");
+const normalizer = require("@ui5/project").normalizer;
+const resourceFactory = require("@ui5/fs").resourceFactory;
+
+
+module.exports = (options) => {
+
+    log.verbose(JSON.stringify(options, undefined, 2));
+
+    const filter = (id) => !options.exclude.includes(id);
+
+    return {
+
+        name: "ui5-resolve",
+
+        buildStart: async (buildOptions) => {
+            const project = await normalizer.generateProjectTree({
+                cwd: options.projectPath
+            });
+            this.dependencies = resourceFactory.createCollectionsForTree(project, {}).dependencies;
+
+            const resources = await this.dependencies.byGlob("/resources/sap/ui/fl/**");
+            const writePromises = resources.map(resource => {
+                const file = `./dist${resource.getPath()}`;
+                const folder = path.dirname(file);
+                if (!fs.existsSync(folder)) {
+                    fs.mkdirSync(folder, { recursive: true });
+                }
+                if (fs.existsSync(file)) {
+                    fs.unlinkSync(file);
+                }
+                return pipe(resource.getStream(), fs.createWriteStream(file));
+            });
+            await Promise.all(writePromises);
+        },
+
+
+        // Right before writting result to dist
+        renderChunk: async (code) => {
+            return "global.window = {};" + code;
```

Comment:
why is this needed?

Reply:
src/sap.ui.core/src/sap/base/util/ObjectPath.js requires a window object. I see 2 possible solutions, either .replace(/window/g, "{}") or add "window = {};" on top. The first solution can lead to unpredicted replacements, so I went with solution 2. Maybe you have a better idea how to handle it?